mne_nirs/utils/_io.py

```
@@ -3,25 +3,91 @@
+import nilearn
 
 
-def _GLM_to_tidy_long(data, labels, glm_estimates, design_matrix):
+def GLM_to_tidy_long(raw, statistic, design_matrix):
+    """
+    Exports GLM regression or contrast results in tidy format.
 
-    theta_estimates = np.zeros((len(labels), len(design_matrix.columns)))
-    t_estimates = np.zeros((len(labels), len(design_matrix.columns)))
-    df_estimates = np.zeros((len(labels), len(design_matrix.columns)))
-    p_estimates = np.zeros((len(labels), len(design_matrix.columns)))
-    mse_estimates = np.zeros((len(labels), len(design_matrix.columns)))
+    Creates a long pandas data frame from regression results or contrast
+    as computed by run_GLM or compute_contrast.
 
-    for idx, lab in enumerate(labels):
-        matching_labels = np.where(([lab == l for l in labels]))
-        matching_idx = np.where([idx == ml for ml in matching_labels])[1]
-        theta_estimates[idx, :] = glm_estimates[lab].theta[:, matching_idx].T
-        df_estimates[idx, :] = glm_estimates[lab].df_model
-        mse_estimates[idx, :] = glm_estimates[lab].MSE[0]
+    Parameters
+    ----------
+    raw : MNE.Raw
+        Instance of MNE raw.
+    statistic : nilearn data,
+        Either dict of nilearn.stats.regression.RegressionResults as returned
+        by run_GLM, or nilearn.stats.contrasts.Contrast as returned by
+        compute_contrast.
+    design_matrix : DataFrame
+        As specified in Nilearn
+
+    Returns
+    -------
+    df : Tidy data frame,
+        Data from statistic object in tidy data form.
+    """
+
+    if isinstance(statistic, dict):
+        if isinstance(statistic[list(statistic.keys())[0]],
+                      nilearn.stats.regression.RegressionResults):
+            df = _tidy_RegressionResults(raw, statistic, design_matrix)
+
+    elif isinstance(statistic, nilearn.stats.contrasts.Contrast):
+        df = _tidy_Contrast(raw, statistic, design_matrix)
+
+    else:
+        warn("Unknown GLM result type")
```

If you hit this code path, df will not be defined when `return df` is called

I would raise an error here instead